### PR TITLE
[WIP] Status Resource

### PR DESCRIPTION
**Pull request by onyiny-ang**

### CODE REVIEW

File: pkg/controller/sync/controller.go [hunk 1]
```diff
@@ -476,6 +484,46 @@ func (s *FederationSyncController) reconcile(qualifiedName util.QualifiedName) u
 		return util.StatusError
 	}
 
+	if targetKind == "FederatedReplicaSet" {
+		status := &frstatv1a1.FederatedReplicaSetStatus{}
```
Comment (pmorie) [inline pkg/controller/sync/controller.go]: Per our discussion on slack, this should not create the `FederatedReplicaSetStatus`, but instead try to get a `FederatedReplicaSetStatus` with the name/namespace of the `FederatedReplicaSet`. If there is no status resource, there's no work to do.

File: pkg/controller/sync/controller.go [hunk 2]
```diff
@@ -476,6 +484,46 @@ func (s *FederationSyncController) reconcile(qualifiedName util.QualifiedName) u
...
+		type clusterStat struct {
+			ClusterName   string
+			Replicas      int32
+			ReadyReplicas int32
+		}
+
+		clusters := []frstatv1a1.FederatedClusterStatus{}
+
+		for cluster := range clusterNames {
+			repSet, err := s.fedClient.CoreV1alpha1().FederatedReplicaSets(qualifiedName.Namespace).Get(clusterNames[cluster], metav1.GetOptions{})
```
Comment (pmorie) [inline pkg/controller/sync/controller.go]: This gets the FederatedReplicaSet with a name you don't expect - instead, this should get the ReplicaSet from each cluster.

Comment (onyiny-ang) [reply pkg/controller/sync/controller.go]: I'm not sure that I fully understand how to get the desired ReplicaSet here.

**Comment by onyiny-ang** [comment]
Attempted to fix the issues with the sync controller and added some test scripts to try to test the federatedreplicasetstatus but I don't think I'm doing it right--or the status resource/controller is not working properly _or_ all of the above :cold_sweat:  The federated resource yamls have changed since the last time

**Pull request closed without merging**